ts/modules/stock-tools-gui.ts

```diff
@@ -934,69 +936,59 @@ addEvent(H.Chart, 'redraw', function (): void {
  * @param {Chart} - Reference to chart
  */
 
-H.Toolbar = function (
-    this: Highcharts.Toolbar,
-    options: Highcharts.StockToolsGuiOptions,
-    langOptions: (Highcharts.Dictionary<string>|undefined),
-    chart: Highcharts.Chart
-): void {
-    this.chart = chart;
-    this.options = options;
-    this.lang = langOptions;
-    // set url for icons.
-    this.iconsURL = this.getIconsURL();
-    this.guiEnabled = options.enabled;
-    this.visible = pick(options.visible, true);
-    this.placed = pick(options.placed, false);
-
-    // General events collection which should be removed upon destroy/update:
-    this.eventsToUnbind = [];
-
-    if (this.guiEnabled) {
-        this.createHTML();
-
-        this.init();
+class Toolbar {
+    public constructor(
+        options: Highcharts.StockToolsGuiOptions,
+        langOptions: (Highcharts.Dictionary<string>|undefined),
+        chart: Highcharts.Chart
+    ) {
+        this.chart = chart;
+        this.options = options;
+        this.lang = langOptions;
+        // set url for icons.
+        this.iconsURL = this.getIconsURL();
+        this.guiEnabled = options.enabled;
+        this.visible = pick(options.visible, true);
+        this.placed = pick(options.placed, false);
+
+        // General events collection which should be removed upon
+        // destroy/update:
+        this.eventsToUnbind = [];
+
+        if (this.guiEnabled) {
+            this.createHTML();
+
+            this.init();
+
+            this.showHideNavigatorion();
+        }
 
-        this.showHideNavigatorion();
+        fireEvent(this, 'afterInit');
     }
 
-    fireEvent(this, 'afterInit');
-} as any;
+    public arrowDown: Highcharts.HTMLDOMElement = void 0 as any;
+    public arrowUp: Highcharts.HTMLDOMElement = void 0 as any;
+    public arrowWrapper: Highcharts.HTMLDOMElement = void 0 as any;
+    public chart: Highcharts.Chart = void 0 as any;
+    public eventsToUnbind: Array<Function> = void 0 as any;
+    public guiEnabled: (boolean|undefined) = void 0 as any;
+    public iconsURL: string = void 0 as any;
+    public lang: (Highcharts.Dictionary<string>|undefined) = void 0 as any;
+    public listWrapper: Highcharts.HTMLDOMElement = void 0 as any;
+    public options: Highcharts.StockToolsGuiOptions = void 0 as any;
+    public placed: boolean =void 0 as any;
+    public showhideBtn: Highcharts.HTMLDOMElement = void 0 as any;
+    public submenu: Highcharts.HTMLDOMElement = void 0 as any;
+    public toolbar: Highcharts.HTMLDOMElement= void 0 as any;
+    public visible: boolean = void 0 as any;
```

Comment:
Properties, that get right away a value in the constructor, should not be initialized (`void 0 as any`). As far as I see, this has to change for `chart`, `eventsToUnbind`, `guiEnabled`, `iconsURL`, `lang`, `options`, `placed`, `visible`.